docbrown/db/src/algorithms/degree.rs

```diff
@@ -0,0 +1,99 @@
+use crate::graph_window::WindowedGraph;
+
+pub fn max_out_degree(windowed_graph: &WindowedGraph) -> usize {
```

Comment:
We can make these more functional and remove some of the calls into the graph, can go through it with you tomorrow.